doc/tutorial/linear-algebra/tutorial-basic-linear-algebra.doc

```diff
@@ -0,0 +1,295 @@
+    - Extract the installer from archive
+      \verbatim
+$ cd ~/Download
+$ tar xvzf l_mkl_2020.0.166.tgz
+      \endverbatim
+    - Run the installer and chose an installation location. By default installation location is set to `$HOME/intel`.
+      \verbatim
+$ cd l_mkl_2020.0.166
+$ ./install_GUI.sh
+      \endverbatim
+    - Set `MKLROOT` environment var to be able to detect MKL during cmake configuration. If you keep the default installation location as `$HOME/intel` simply run:
+      \verbatim
+$ echo "export MKLROOT=$HOME/intel/mkl" >> ~/.bashrc
+$ source ~/.bashrc
+      \endverbatim
+- [OpenBLAS](https://www.openblas.net/) an optimized BLAS library
+  - To install openBLAS on Ubuntu proceed with:
+    \verbatim
+$ sudo apt install libopenblas-dev
+    \endverbatim
+- [Netlib](https://www.netlib.org/) a collection of mathematical software
+  - To install Netlib on Ubuntu proceed with:
+    \verbatim
+$ sudo apt install liblapack-dev
+    \endverbatim
+- [Atlas](http://math-atlas.sourceforge.net/) for Automatically Tuned Linear Algebra Software
+  - To install Atlas on Ubuntu proceed with:
+    \verbatim
+$ sudo apt install libatlas-base-dev
+    \endverbatim
+- [GSL](http://math-atlas.sourceforge.net/) for GNU Scientific Library, that only implements the C interface, not the Fortran interface 
+  - To install GSL on Ubuntu proceed with:
+    \verbatim
+$ sudo apt install libgsl-dev
+    \endverbatim
+
+If none of these third-parties is installed, ViSP provides a build-in version of Lapack that is not optimized and that could be used instead. Note that depending on our test performances some linear algebra operations may use lapack built-in or naive code when we found that it runs faster.
+
+\note
+- When at least one of the following third-party between `MKL`, `OpenBLAS`, `Netlib`,  `Atlas` or `GSL` is installed `VISP_HAVE_LAPACK` macro is defined in `vpConfig.h` file.
+There is also a more specific macro that is defined with the used 3rd party as suffix, `VISP_HAVE_LAPACK_MKL`, `VISP_HAVE_LAPACK_OPENBLAS`, `VISP_HAVE_LAPACK_NETLIB`, `VISP_HAVE_LAPACK_ATLAS` and `VISP_HAVE_LAPACK_GSL` respectively.
+- When none of these third-parties is used, as explained previously lapack build-in is used and `VISP_HAVE_LAPACK` and `VISP_HAVE_LAPACK_BUILD_IN` macro are then defined.
+- During cmake configuration you will be asked to select one of these third-party
+  \verbatim
+$ cd $VISP_WS/visp-build
+$ ccmake ../visp
+  \endverbatim
+  At this step, selection is achieved tuning the following line:
+  \verbatim
+USE_BLAS/LAPACK           [ MKL | OpenBLAS | Netlib | Atlas | GSL ]
+  \endverbatim
+- Note also that if you select `OpenBLAS`, or `Netlib` or `Atlas`, there is the possibility to change the used third-party during run time, without re-building ViSP. On Ubuntu, this could be achieved using `update-alternatives`:
+  \verbatim
+$ sudo update-alternatives --config liblapack.so.3-x86_64-linux-gnu
```

Comment:
I need to check my benchmark results again using `update-alternatives` on `libblas` since I only change `liblapack` alternative!

-----
There is something strange in my Ubuntu 18.04:
```
$ update-alternatives --config liblapack.so.3
update-alternatives: error: no alternatives for liblapack.so.3
``` 
I need to run:
```
update-alternatives --config liblapack.so.3-x86_64-linux-gnu
There are 3 choices for the alternative liblapack.so.3-x86_64-linux-gnu (providing /usr/lib/x86_64-linux-gnu/liblapack.so.3).

  Selection    Path                                               Priority   Status
------------------------------------------------------------
  0            /usr/lib/x86_64-linux-gnu/openblas/liblapack.so.3   40        auto mode
  1            /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3      35        manual mode
  2            /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3     10        manual mode
* 3            /usr/lib/x86_64-linux-gnu/openblas/liblapack.so.3   40        manual mode
```

Same for `libblas`
```
$ update-alternatives --config libblas.so.3
update-alternatives: error: no alternatives for libblas.so.3

$ update-alternatives --config libblas.so.3-x86_64-linux-gnu
There are 3 choices for the alternative libblas.so.3-x86_64-linux-gnu (providing /usr/lib/x86_64-linux-gnu/libblas.so.3).

  Selection    Path                                             Priority   Status
------------------------------------------------------------
* 0            /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3   40        auto mode
  1            /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3      35        manual mode
  2            /usr/lib/x86_64-linux-gnu/blas/libblas.so.3       10        manual mode
  3            /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3   40        manual mode
```

Reply:
See https://wiki.debian.org/DebianScience/LinearAlgebraLibraries
Since buster it is:
```
update-alternatives --config libblas.so.3-<multiarch>
update-alternatives --config liblapack.so.3-<multiarch>
```